hta/common/trace_parser.py

```diff
@@ -249,6 +249,13 @@ def _compress_df(
     """
     cfg = cfg or ParserConfig.get_default_cfg()
 
+    # Keep memory events (which are instant events)
+    is_memory_event = (
+        (df["ph"] == "i")
+        & (df["name"] == "[memory]")
+        & (df["cat"] == "cpu_instant_event")
+    )
+    df.loc[is_memory_event, "dur"] = 0
```

Comment:
qq we are just turning the "dur"  to 0 here , how does that keep the memoy events, i thought they are already retained

Reply:
Memory events have no duration as they are "instantaneous" - as a consequence "dur" is set to `na` but the following lines do:

```python
    # drop rows with null values
    df.dropna(axis=0, subset=["dur", "cat"], inplace=True)
```

Which dropped them.